inductiva/resources/machines.py

```diff
@@ -0,0 +1,119 @@
+        self.name = self._generate_instance_name()
+        self.machine_group_id = resources.create_machine_group_id()
+        self.zone = zone
+        self.api_config = api.validate_api_key(inductiva.api_key)
+
+    def start(self):
+        with ApiClient(self.api_config) as client:
+            api_instance = InstanceApi(client)
+
+            body = InstanceGroup(
+                name=self.name,
+                machine_type=self.machine_type,
+                num_instances=self.num_machines,
+                spot=self.spot,
+                resource_pool_id=self.machine_group_id,
+                disk_size_gb=self.disk_size_gb,
+                zone=self.zone,
+            )
+            # body.resource_pool_id = self.machine_group_id
+            try:
+                logging.info("Creating a machine group. \
+                             This may take a few minutes.")
+                api_instance.create_instance_group(body=body)
+                logging.info(
+                    "Machine group with the specified settings successfully \
+                        created.")
+                logging.info("Machine type: %s", self.machine_type)
+                logging.info("Number of machines: %s", self.num_machines)
+                logging.info("Spot: %s", self.spot)
+                logging.info("Disk size: %s GB", self.disk_size_gb)
+                logging.info("Machine group ID: %s", self.machine_group_id)
+                if self.name:
+                    logging.info("Name: %s", self.name)
+                logging.info("Machine group successfully created.")
+
+            except ApiException as e:
+                raise e
+        return self.machine_group_id
+
+    def terminate(self):
+
+        with ApiClient(self.api_config) as client:
+            api_instance = InstanceApi(client)
+
+            try:
+                logging.info("Terminating machine group. \
+                             This may take a few minutes.")
+                api_instance.delete_instance_group(body=Instance(
+                    name=self.name))
+                logging.info(
+                    "Machine group of %s machines successfully \
+                    terminated.", self.num_machines)
+
+            except ApiException as e:
+                raise e
+
+    def estimate_cost(self):
+        """Returns an estimated cost per hour of a machine group."""
+        with ApiClient(self.api_config) as client:
+            api_instance = InstanceApi(client)
+
+            try:
+                instance_price = api_instance.get_instance_price(
+                    body=Instance(name=self.machine_type, zone=self.zone))
+                logging.info(
+                    "Estimated cost of a machine group is %s$ per hour.",
+                    self._get_cost(instance_price))
+            except ApiException as e:
+                raise e
+        return self._get_cost(instance_price)
+
+    def _get_cost(self, instance_price):
```

Comment:
is this _compute_cost? Also, why do you need this function? Why not directly integrate this in the previous function.